Objects/Objects/BuiltElements/Rebar.cs

```diff
@@ -57,6 +210,88 @@ public enum shape
 
 namespace Objects.BuiltElements.Revit
 {
+  public class RevitRebarGroup : RebarGroup
+  {
+    public RevitRebarGroup() { }
+
+    public override RebarShape shape
+    {
+      get => revitShape;
+      set
+      {
+        if (value is not RevitRebarShape s)
+        {
+          throw new ArgumentException($"Expected object of type {nameof(RevitRebarShape)}");
+        }
+
+        revitShape = s;
+      }
+    }
```

Comment:
This is a bit tricky. C# 9 supports covarience on return types, but I think we're stuck on 8 in objects. Did you consider generic properties instead?
```
public class RebarGroup<T> where T : RebarShape
{
  public T shape { get; set; }
}
```
You'd get more type safety that way, but I'm actually not sure how well our serializer handles generics.

Reply:
This seems like a good suggestion - will implement this and remove the new `TeklaRebarGroup` class, since I haven't scoped a `TeklaRebarShape` class yet

Reply:
I don't think we are on 8, Alan took a look and seems to be 11...https://github.com/specklesystems/speckle-sharp/blob/348fe44cd379a56cbdaf6da8454300271e88826b/Directory.Build.props#L17C3-L18C26

Reply:
But I like Connor's suggestion, but...  so long as we are serialising a RevitRebarGroup Alan believes this should be fine with the serialiser